src/FFmpegReader.cpp

```diff
@@ -1315,8 +1319,23 @@ void FFmpegReader::ProcessVideoPacket(int64_t requested_frame) {
 		if (openshot::Settings::Instance()->HIGH_QUALITY_SCALING) {
 			scale_mode = SWS_LANCZOS;
 		}
-		SwsContext *img_convert_ctx = sws_getContext(info.width, info.height, AV_GET_CODEC_PIXEL_FORMAT(pStream, pCodecCtx), width,
-															  height, PIX_FMT_RGBA, scale_mode, NULL, NULL, NULL);
+		
+		// In case of 4:4:4 inputs, FFmpeg scaling optimization for fast_bilinear
+		// makes undesirable horizontal chroma simplifications.
+		// Do not reuse chroma for 2 pixels RGB/BGR in this case.
+		scale_mode |= SWS_FULL_CHR_H_INT;
+
+		SwsContext *img_convert_ctx = sws_getContext(info.width, info.height, pCodecCtx->pix_fmt, width,
```

Comment:
pCodecCtx-pix_fmt should be `AV_GET_CODEC_PIXEL_FORMAT(pStream, pCodecCtx)`. This allows us to build against other versions of LibAV and FFmpeg.